### ISSUE TITLE
Selection of videos

### ISSUE THREAD

**Issue by zhanghousen** [lordmartian/deep_avsr#26]
Dear Professor, is it possible to get all the data in each training step when training the network?
Because in the code I see that the videos of pretrain and train are grouped and only one video of the group is selected at a time.

if self.dataset == "pretrain":
            #（1）index goes from 0 to stepSize-1
            #（2）dividing the dataset into partitions of size equal to stepSize and selecting a random partition
            #（3）fetch the sample at position 'index' in this randomly selected partition
            base = self.stepSize * np.arange(int(len(self.datalist)/self.stepSize)+1)  
            ixs = base + index
            ixs = ixs[ixs < len(self.datalist)]
            index = np.random.choice(ixs)    
        targetFile = self.datalist[index] + ".txt"
        targetFile_name = self.datalist[index]
        inp, trgt, inpLen, trgtLen = prepare_pretrain_input(visualFeaturesFile, targetFile, self.numWords, self.charToIx, self.videoParams)

Is not grouped during validation and testing？

In addition, this model is TM-CTC, could the TM-seq2seq model be published.

Hope to get your answer

**Comment by lordmartian** [comment]
Hello,
The grouping is done only for pretrain/train datasets. During validation/testing, we compute loss and error rates over the entire validation/test dataset. You can take entire pretrain/train dataset every step by setting step size equal to the size of pretrain/train dataset. However, you may need to alter the hyperparameter values to obtain the best results.
We have trained only the TM-CTC model and not the TM-seq2seq model.

**Comment by zhanghousen** [comment]
Thank you for your reply.
